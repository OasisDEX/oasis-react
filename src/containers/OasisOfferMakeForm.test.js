/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { fromJS } from 'immutable';

import {
  OasisOfferMakeFormWrapper,
  mapStateToProps,
  mapDispatchToProps
} from './OasisOfferMakeForm';
import { shallow } from 'enzyme';

describe('(Container) OasisOfferMakeForm', () => {
  const state = fromJS(global.storeMock);
  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });


  it('will receive right actions', () => {
    expect(initialActions).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(
      <OasisOfferMakeFormWrapper {...props}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
