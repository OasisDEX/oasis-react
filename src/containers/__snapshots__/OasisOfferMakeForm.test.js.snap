// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) OasisOfferMakeForm should render 1`] = `
<Connect(ReduxForm)
  actions={
    Object {
      "buyMax": [Function],
      "priceFieldValueChanged": [Function],
      "sellMax": [Function],
      "totalFieldValueChanged": [Function],
      "volumeFieldValueChanged": [Function],
    }
  }
  activeTradingPairPrecision={18}
  currentFormValues={Object {}}
  hasSufficientTokenAmount={false}
  offerMakeType="OFFER_MAKES/MAKE_BUY_OFFER"
/>
`;

exports[`(Container) OasisOfferMakeForm will receive right actions 1`] = `
Object {
  "actions": Object {
    "buyMax": [Function],
    "priceFieldValueChanged": [Function],
    "sellMax": [Function],
    "totalFieldValueChanged": [Function],
    "volumeFieldValueChanged": [Function],
  },
}
`;

exports[`(Container) OasisOfferMakeForm will receive right props 1`] = `
Object {
  "activeBaseTokenBalance": undefined,
  "activeQuoteTokenBalance": undefined,
  "activeTradingPairPrecision": 18,
  "currentFormValues": Object {},
  "hasSufficientTokenAmount": false,
}
`;
